accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicMonotonicTestForAlternateRanges.java

```diff
@@ -0,0 +1,121 @@
+import uk.ac.ebi.ampt2d.commons.accession.service.BasicMonotonicAccessioningService;
+import uk.ac.ebi.ampt2d.test.TestModel;
+import uk.ac.ebi.ampt2d.test.configuration.TestMonotonicDatabaseServiceTestConfiguration;
+import uk.ac.ebi.ampt2d.test.service.TestMonotonicDatabaseService;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@ContextConfiguration(classes = {TestMonotonicDatabaseServiceTestConfiguration.class})
+@TestPropertySource("classpath:application-test1.properties")
+public class BasicMonotonicTestForAlternateRanges {
+
+    @Autowired
+    private TestMonotonicDatabaseService databaseService;
+
+    @Autowired
+    private ContiguousIdBlockService contiguousIdBlockService;
+
+    @Autowired
+    @Qualifier("getMonotonicGeneratorPropertiesForEva")
+    private MonotonicGeneratorProperties monotonicGeneratorProperties;
+
+    @Test
+    public void testAlternateRangesWithPrefixes() throws AccessionCouldNotBeGeneratedException {
+        Map<String, TestModel> objects = new LinkedHashMap<>();
+        objects.put("hash1", TestModel.of("service-test-1"));
+        objects.put("hash2", TestModel.of("service-test-2"));
+        objects.put("hash3", TestModel.of("service-test-3"));
+        objects.put("hash4", TestModel.of("service-test-4"));
+        objects.put("hash5", TestModel.of("service-test-5"));
+        objects.put("hash6", TestModel.of("service-test-6"));
+        DecoratedAccessionGenerator<TestModel, Long> generator = DecoratedAccessionGenerator
+                .buildPrefixSuffixAccessionGenerator(getGenerator(), "RS", null, Long::parseLong);
+        List<AccessionWrapper<TestModel, String, String>> generated = generator.generateAccessions(objects);
+        assertEquals(6, generated.size());
+        assertEquals("RS1", generated.get(0).getAccession());
+        assertEquals("RS2", generated.get(1).getAccession());
+        assertEquals("RS11", generated.get(5).getAccession());
+
+    }
+
+    @Test
+    public void testAlternateRangesWithDifferentBlockSizes() throws AccessionCouldNotBeGeneratedException {
+        monotonicGeneratorProperties.setBlockSize(5);
+        monotonicGeneratorProperties.setNextBlockInterval(5);
+        List<AccessionWrapper<TestModel, String, Long>> evaAccessions = getAccessioningService1().getOrCreate
+                (IntStream.range(1, 21).mapToObj(i -> TestModel.of("Test-" + i)).collect(Collectors.toList()));
+        assertEquals(20, evaAccessions.size());
+        assertEquals(35, evaAccessions.get(19).getAccession().longValue());
+        assertFalse(evaAccessions.stream().anyMatch(accession -> accession.getAccession() % 10 > 5));
```

Comment:
Why not running this check in later stages of the method?

Reply:
added clear validations than before